packages/@sanity/desk-tool/src/panes/document/useDocumentIsReferenced.tsx

```diff
@@ -0,0 +1,13 @@
+import {createHookFromObservableFactory, getPublishedId} from '@sanity/base/_internal'
+import documentStore from 'part:@sanity/base/datastore/document'
+import {Observable} from 'rxjs'
+
+export const useDocumentsIsReferenced = createHookFromObservableFactory((documentId: string) => {
+  const totalClause = 'count(*[references($documentId)]) > 0'
```

Comment:
Will this work for referencing documents that you don't have permissions for?

Reply:
I don't know.. @rexxars, do you know?

Reply:
Good point - it does not, unfortunately. I've escalated this internally, we'll discuss further there.